dissect/target/plugins/filesystem/ntfs/mft.py

```diff
@@ -138,53 +145,62 @@ def mft(self, compact: bool = False):
         else:
             record_formatter = formatter
 
-        for fs in self.target.filesystems:
-            if fs.__type__ != "ntfs":
-                continue
-
-            # If this filesystem is a "fake" NTFS filesystem, used to enhance a
-            # VirtualFilesystem, The driveletter (more accurate mount point)
-            # returned will be that of the VirtualFilesystem. This makes sure
-            # the paths returned in the records are actually reachable.
-            drive_letter = get_drive_letter(self.target, fs)
-            volume_uuid = get_volume_identifier(fs)
-
+        if fs:
             try:
-                for record in fs.ntfs.mft.segments():
-                    try:
-                        inuse = bool(record.header.Flags & FILE_RECORD_SEGMENT_IN_USE)
-                        owner, _ = get_owner_and_group(record, fs)
-                        resident = False
-                        size = None
-
-                        if not record.is_dir():
-                            for data_attribute in record.attributes.DATA:
-                                if data_attribute.name == "":
-                                    resident = data_attribute.resident
-                                    break
-
-                            size = get_record_size(record)
-
-                        for path in record.full_paths():
-                            path = f"{drive_letter}{path}"
-                            yield from self.mft_records(
-                                drive_letter=drive_letter,
-                                record=record,
-                                segment=record.segment,
-                                path=path,
-                                owner=owner,
-                                size=size,
-                                resident=resident,
-                                inuse=inuse,
-                                volume_uuid=volume_uuid,
-                                record_formatter=record_formatter,
-                            )
-                    except Exception as e:
-                        self.target.log.warning("An error occured parsing MFT segment %d: %s", record.segment, str(e))
-                        self.target.log.debug("", exc_info=e)
-
-            except Exception:
-                log.exception("An error occured constructing FilesystemRecords")
+                filesystem = self.ntfs_filesystems[fs]
+            except IndexError:
+                self.target.log.exception("%s does not have a filesystem with index number: %s", self.target, fs)
```

Comment:
This error message might be confusing for the user, because `filesystems` is a subset of the available filesystems. Therefore, the index numbering of `filesystems` may differ from the actual filesystems. Maybe a better alternative is to make `filesystems` a `dict` where the indices are the keys and filesystems objects are the values?

Reply:
Fixed